src/index.js

```diff
@@ -1,203 +1,201 @@
-import postcss from 'postcss'
-import Tokenizer from 'css-selector-tokenizer'
-import { extractICSS, createICSSRules } from 'icss-utils'
-
-let hasOwnProperty = Object.prototype.hasOwnProperty
-
-function getSingleLocalNamesForComposes(selectors) {
-  return selectors.nodes.map(node => {
-    if (node.type !== 'selector' || node.nodes.length !== 1) {
-      throw new Error(
-        'composition is only allowed when selector is single :local class name not in "' +
-          Tokenizer.stringify(selectors) +
-          '"'
-      )
-    }
-    node = node.nodes[0]
-    if (
-      node.type !== 'nested-pseudo-class' ||
-      node.name !== 'local' ||
-      node.nodes.length !== 1
-    ) {
-      throw new Error(
-        'composition is only allowed when selector is single :local class name not in "' +
-          Tokenizer.stringify(selectors) +
-          '", "' +
-          Tokenizer.stringify(node) +
-          '" is weird'
-      )
-    }
-    node = node.nodes[0]
-    if (node.type !== 'selector' || node.nodes.length !== 1) {
-      throw new Error(
-        'composition is only allowed when selector is single :local class name not in "' +
-          Tokenizer.stringify(selectors) +
-          '", "' +
-          Tokenizer.stringify(node) +
-          '" is weird'
-      )
-    }
-    node = node.nodes[0]
-    if (node.type !== 'class') {
-      // 'id' is not possible, because you can't compose ids
-      throw new Error(
-        'composition is only allowed when selector is single :local class name not in "' +
-          Tokenizer.stringify(selectors) +
-          '", "' +
-          Tokenizer.stringify(node) +
-          '" is weird'
-      )
-    }
-    return node.name
-  })
-}
-
-const defaultGenerateScopedName = function(exportedName, path) {
-  let sanitisedPath = path
-    .replace(/\.[^\.\/\\]+$/, '')
-    .replace(/[\W_]+/g, '_')
-    .replace(/^_|_$/g, '')
-  return `_${sanitisedPath}__${exportedName}`
-}
-
-module.exports = postcss.plugin(
-  'postcss-modules-scope',
-  (options = {}) => css => {
-    let generateScopedName =
-      options.generateScopedName || defaultGenerateScopedName
-
-    let exports = {}
-
-    function exportScopedName(name) {
-      let scopedName = generateScopedName(
-        name,
-        css.source.input.from,
-        css.source.input.css
-      )
-      exports[name] = exports[name] || []
-      if (exports[name].indexOf(scopedName) === -1) {
-        exports[name].push(scopedName)
-      }
-      return scopedName
+/* eslint-env node */
+import postcss from "postcss";
+import Tokenizer from "css-selector-tokenizer";
```

Comment:
https://github.com/webpack-contrib/css-loader/pull/523 Maybe in the long-term better replaced by e.g `postcss-selector-parser` but not _sooo_ important now 😛